src/ocean/net/http/HttpServer.d

```diff
@@ -0,0 +1,232 @@
+
+/// The Http configuration
+public struct HttpConfig
+{
+    /// Address used for listening for new connections
+    public string address;
+
+    /// Port used for listening for new connections
+    public ushort port;
+}
+
+/// Http server that handle generic requests using a list of handlers
+public class HttpServer
+{
+    /// Logger
+    protected Logger log;
+
+    /// The ipv4 socket used by the web server
+    private AddressIPSocket!(false) socket;
+
+    /// The ipv4 address used to setup the socket
+    private InetAddress!(false) address;
+
+    /// Http server used to serve prometheus stats and other inputs if it's needed
+    alias HttpServerListener = SelectListener!(HttpRequestHandler, HttpHandler[]);
+
+    /// ditto
+    private HttpServerListener http_server;
+
+    /// The list of request handlers
+    private HttpHandler[] handlers;
+
+    /**************************************************************************
+
+        Constructor
+
+        Params:
+            handlers = The list of http handlers that the server will use
+
+    **************************************************************************/
+
+    public this ( HttpHandler[] handlers )
+    {
+        this.log = Log.lookup("ocean.net.http.HttpServer");
+        this.handlers = handlers;
+    }
+
+    /**************************************************************************
+
+        Start the http server
+
+        Params:
+            config = The http configuration
+
+    **************************************************************************/
+
+    public void start ( HttpConfig config )
+    {
+        log.info("Listening on {}:{}", config.address, config.port);
+        auto listen_address = address(config.address, config.port);
+        enforce(listen_address !is null, "Can't listen for TCP connections.");
+
+        this.socket = new AddressIPSocket!();
+        this.http_server = new HttpServerListener(listen_address, this.socket, this.handlers);
+        log.info("Waiting HTTP requests on {}:{}", config.address, config.port);
+
+        theScheduler.epoll.register(this.http_server);
+    }
+
+    /// Stop listening for new requests
+    public void terminate ( )
+    {
+        this.http_server.terminate();
+        theScheduler.epoll.unregister(this.http_server);
+    }
+}
+
+/// Handle the list of requests in order until one of them returns something
+/// else than `Not Found`
+private class HttpRequestHandler : TaskHttpConnectionHandler
+{
+    /// The http handler list
+    private HttpHandler[] handlers;
+
+    /// Log instance
+    private Logger log;
+
+    /**************************************************************************
+
+        Constructor
+
+        Params:
+            finalizer = finalizer callback of the select listener
+
+    **************************************************************************/
+
+    public this ( scope FinalizeDg finalizer, HttpHandler[] handlers )
```

Comment:
You really don't want to use `scope` here

Reply:
weird... it's used like this everywhere in the released version. It compiles without it, so I removed `scope`

https://github.com/sociomantic-tsunami/ocean/blob/v5.1.0/integrationtest/selectlistener/UnixServer.d#L102

https://github.com/sociomantic-tsunami/ocean/blob/v5.1.0/src/ocean/net/http/HttpConnectionHandler.d#L125

https://github.com/sociomantic-tsunami/ocean/blob/v5.1.0/src/ocean/net/server/connection/IConnectionHandler.d#L161

https://github.com/sociomantic-tsunami/ocean/blob/v5.1.0/src/ocean/util/prometheus/server/PrometheusHandler.d#L66

and so on....